### PR TITLE
CBPB manure module: Nitrogen gas emission calculations

### PR DESCRIPTION
**Pull request by michaelRichards99**
Implements the 3 gas calculations that combine to calculate the total loss of nitrogen in the manure-bedding mixture for the CBPB manure module.

This PR introduces 4 functions:
1. `_calc_nitrogen_loss_from_ammonia_emissions()`: M.3.A.1
2. `_calc_nitrogen_loss_to_leaching()`: M.3.B.1
3. `_calc_nitrogen_loss_from_nitrous_oxide_emissions()`: M.3.C.1
4. `calc_nitrogen_losses()`: M.3.D.1. This only includes the losses. I will add the additional N inside the CBPB class.

Please refer to the CBPB design doc for the equations.

### CODE REVIEW

**Review by ltp23 — APPROVED** [review]
If you have time, maybe you could add more error handling for the parameters being passed into those equation functions and update the tests to include more edge cases. 

Unless Varma wants to make changes to the equations, I'm ok with what we have here. 

Thanks Michael. Good job!

File: RUFAS/routines/manure/gas_emissions/gas_emissions.py [hunk 1]
```diff
@@ -1295,3 +1295,116 @@ def calc_total_carbon_decomposition(
...
+        existing_nitrogen_mass : float
+            The mass of nitrogen already present in the manure-bedding mixture (kg)
+
+        Returns
+        -------
+        float
+            The amount of nitrogen that leaches out of the bedding mixture (kg).
+
+        """
+        return 0.035 * (daily_nitrogen_input + existing_nitrogen_mass)
```
Comment (varrma) [inline RUFAS/routines/manure/gas_emissions/gas_emissions.py]: You can remove the existing_nitrogen_mass; as discussed we account for daily N.

File: RUFAS/routines/manure/gas_emissions/gas_emissions.py [hunk 2]
```diff
@@ -1295,3 +1295,116 @@ def calc_total_carbon_decomposition(
...
+            Indicator for if the beddint is tilled for the current simulation day.
+
+        Returns
+        -------
+        float
+            The nitrogen lost to nitrous oxide emissions (kg).
+
+        """
+        till_indicator = int(is_bedding_tilled)
+        total_nitrogen = daily_nitrogen_input + existing_nitrogen_mass
```
Comment (varrma) [inline RUFAS/routines/manure/gas_emissions/gas_emissions.py]: daily nitrogen input is enough

File: RUFAS/routines/manure/gas_emissions/gas_emissions.py [hunk 3]
```diff
@@ -1295,3 +1295,116 @@ def calc_total_carbon_decomposition(
...
+            Indicator for if the beddint is tilled for the current simulation day.
+
+        Returns
+        -------
+        float
+            The nitrogen lost to ammonia emissions (in kg).
+
+        """
+        till_indicator = int(is_bedding_tilled)
+        total_nitrogen = daily_nitrogen_input + existing_nitrogen_mass
```
Comment (varrma) [inline RUFAS/routines/manure/gas_emissions/gas_emissions.py]: please remove the "existing_nitrogen_mass" - as we consider the daily N mass

**Review by varrma — APPROVED** [review]
Good work Michael, few changes. Rest looks good.

**Comment by michaelRichards99** [comment]
> If you have time, maybe you could add more error handling for the parameters being passed into those equation functions and update the tests to include more edge cases.
> 
> Unless Varma wants to make changes to the equations, I'm ok with what we have here.
> 
> Thanks Michael. Good job!

I've added a check for negative mass on the `calc_nitrogen_losses` function, which is the entry point.
